src/plugins/query_enhancements/server/search/ppl_async_search_strategy.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ILegacyClusterClient, Logger, SharedGlobalConfig } from 'opensearch-dashboards/server';
+import { Observable } from 'rxjs';
+import {
+  createDataFrame,
+  DATA_FRAME_TYPES,
+  IDataFrameResponse,
+  IOpenSearchDashboardsSearchRequest,
+  Query,
+} from '../../../data/common';
+import { ISearchStrategy, SearchUsage } from '../../../data/server';
+import { buildQueryStatusConfig, getFields, handleFacetError, SEARCH_STRATEGY } from '../../common';
+import { Facet } from '../utils';
+
+export const pplAsyncSearchStrategyProvider = (
+  config$: Observable<SharedGlobalConfig>,
+  logger: Logger,
+  client: ILegacyClusterClient,
+  usage?: SearchUsage
+): ISearchStrategy<IOpenSearchDashboardsSearchRequest, IDataFrameResponse> => {
+  const pplAsyncFacet = new Facet({
+    client,
+    logger,
+    endpoint: 'enhancements.runDirectQuery',
+  });
+  const pplAsyncJobsFacet = new Facet({
+    client,
+    logger,
+    endpoint: 'enhancements.getJobStatus',
+    useJobs: true,
+  });
+
+  return {
+    search: async (context, request: any, options) => {
+      try {
+        const query: Query = request.body.query;
+        const pollQueryResultsParams = request.body.pollQueryResultsParams;
+        const inProgressQueryId = pollQueryResultsParams?.queryId;
+
+        if (!inProgressQueryId) {
+          request.body = { ...request.body, lang: SEARCH_STRATEGY.PPL };
+          const rawResponse: any = await pplAsyncFacet.describeQuery(context, request);
+
+          if (!rawResponse.success) handleFacetError(rawResponse);
+
+          const statusConfig = buildQueryStatusConfig(rawResponse);
+
+          return {
+            type: DATA_FRAME_TYPES.POLLING,
+            status: 'started',
+            body: {
+              queryStatusConfig: statusConfig,
+            },
+          } as IDataFrameResponse;
+        } else {
+          request.params = { queryId: inProgressQueryId };
+          const queryStatusResponse: any = await pplAsyncJobsFacet.describeQuery(context, request);
```

Comment:
@ps48 should this have
```
if (!queryStatusResponse.success) handleFacetError(queryStatusResponse);
```
otherwise if `queryStatusResponse.success` is false, then `queryStatusResponse.data` is the Error object, not actual data and can't be processed?
see https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8765/files

Reply:
@ps48 will do a separate follow up PR

Reply:
follow up PR is https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8771